pkg/asset/installconfig/aws/session.go

```diff
@@ -197,9 +204,35 @@ func (ar *awsResolver) EndpointFor(service, region string, optFns ...func(*endpo
 			SigningRegion: ar.region,
 		}, nil
 	}
+	if rv, ok := ar.defaultEndpoints[region]; ok {
+		if v, ok := rv[service]; ok {
+			return v, nil
+		}
+	}
 	return endpoints.DefaultResolver().EndpointFor(service, region, optFns...)
 }
 
 func resolverKey(service string) string {
 	return service
 }
+
+// this is a list of known default endpoints for specific regions that would
+// otherwise require user to set the service overrides.
+// it's a map of region => service => resolved endpoint
+// this is only used when the user hasn't specified a override for the service in that region.
```

Comment:
nit: This is the same comment as lines 179-182. It should probably only appear once.

Reply:
hmm.. personally I want to keep it the same for now, I think the internal one at 179-182 will change in the future but this function is going to stay the same.